### ISSUE TITLE
[ENH] in CI, test only lines that are changed by PR

### ISSUE THREAD

**Issue by fkiraly** [alan-turing-institute/sktime#1408]
I think it would be a great improvement if the PR tests only ran the tests that touch changed lines - it would solve our issue with testing taking hours for every PR.

I'm not sure how one would do this as part of the integrated tests, but there seem to be packages like `pytest-testmon` or `pytest-picked` that allow to do this in-principle.

Perhaps someone who has experience with this sees that it's just a few-line change somewhere and can pick this up?

**Comment by mloning** [comment]
I agree that speeding up CI is important. I'm not sure if only running tests for the changed lines is a good idea. Are you aware of any projects that do that? Identifying the relevant tests to run will be tricky I think. 

The main bottleneck at the moment is appveyor (see #1319) and the very slow distance-based classifiers. Also see https://github.com/alan-turing-institute/sktime/issues/1032

**Comment by fkiraly** [comment]
>Identifying the relevant tests to run will be tricky I think.

exactly the tests that previously touched a line that was changed?

>The main bottleneck at the moment is appveyor (see #1319) and the very slow distance-based classifiers. Also see #1032

Yes, but there will always be some monstrosities like that.
The suggestion would get it down to minutes?

**Comment by fkiraly** [comment]
>Are you aware of any projects that do that?

No, but it seems like a good idea?
